### PR TITLE
✨ Inspect correlation features

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Remove unnecessary private properties and use fields instead
2. Add the following functions to the ManagementAPI interface:
  - `getAllCorrelations`: Gets all stored correlations
  - `getActiveCorrelations`: gets all active correlations
  - `getCorrelationById`: Gets a specific correlation by its ID
  - `getCorrelationsByProcessModelId`: Gets all Correlations in which a specific ProcessModel was run
  - `getCorrelationByProcessInstanceId`: Gets the Correlation in which a specific ProcessInstance was run
3. Rename `getLogsForProcessModel` to `getProcessModelLog`
4. Update the Correlation Converter to include the new Correlation properties and structure

PR: #19

## How can others test the changes?

Use the service to request correlation related data.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/converters/correlation_converter.ts [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
   const managementApiCorrelation: Correlation = new Correlation();
   managementApiCorrelation.id = runtimeCorrelation.id;
-  managementApiCorrelation.processModelId = runtimeCorrelation.processModelId;
+  managementApiCorrelation.state = CorrelationState[runtimeCorrelation.state];
+  managementApiCorrelation.identity = runtimeCorrelation.identity;
+  managementApiCorrelation.createdAt = runtimeCorrelation.createdAt;
+
+  managementApiCorrelation.processModels =
+    runtimeCorrelation.processModels.map((runtimeProcessModel: any): any => {
+      const managementApiProcessModel: CorrelationProcessModel = new CorrelationProcessModel();
```
Comment (SteffenKn) [inline src/converters/correlation_converter.ts]: Please add a spacing after this line to improve the readability.

File: src/management_api_service.ts [hunk 2]
```diff
@@ -35,85 +42,78 @@ import * as BluebirdPromise from 'bluebird';
...
 
-  private _consumerApiService: IConsumerApi;
-  private _correlationService: ICorrelationService;
-  private _deploymentApiService: IDeploymentApi;
-  private _kpiApiService: IKpiApiService;
-  private _loggingApiService: ILoggingApiService;
-  private _processModelFacadeFactory: IProcessModelFacadeFactory;
-  private _processModelService: IProcessModelService;
-  private _tokenHistoryApiService: ITokenHistoryApiService;
+  private consumerApiService: IConsumerApi;
```
Comment (SteffenKn) [inline src/management_api_service.ts]: Did you remove the '_' on purpose?

Comment (ElRaptorus) [reply src/management_api_service.ts]: Yes. Because I'm lazy. :3 
I'll readd them.

**Comment by ElRaptorus** [comment]
@SteffenKn done

**Pull request merged**